### PR TITLE
Updated WebXR Hand implementation to the new API

### PR DESCRIPTION
**Pull request by cabanier**
**Description**
The WebXR API has been updated per feedback from the W3C: https://immersive-web.github.io/webxr-hand-input
This breaks the current implementation.

This patch will update three.js so it will work with the new API. However, it will no longer work with the old API.
Since most headsets are updated quickly, this shouldn't cause too many problems.

This patch should only be merged once the API goes live.

This contribution is funded by [Oculus](https://oculus.com).

### CODE REVIEW

**Comment by cabanier** [comment]
@fernandojsg can you take a look?
I can also send you a private build if it helps.

**Comment by fernandojsg** [comment]
@cabanier it looks good to me. The only thing I don't really like is using numbers to refer to the joints, it makes the code harder to read. Can't we just create constants with these names and expose them directly in three even if the WebXR doens't include these constants anymore?

```javascript
// Previously we had
const indexTip = controller.joints[ XRHand.INDEX_PHALANX_TIP ];

// This PR proposal
const indexTip = controller.joints[ 9 ];

// Something more readable ?
const indexTip = controller.joints[ THREE.XRHand.INDEX_PHALANX_TIP ];
```

**Comment by cabanier** [comment]
> @cabanier it looks good to me. The only thing I don't really like is using numbers to refer to the joints, it makes the code harder to read. Can't we just create constants with these names and expose them directly in three even if the WebXR doens't include these constants anymore?
> 
> ```js
> ...
> // Something more readable ?
> const indexTip = controller.joints[ THREE.XRHand.INDEX_PHALANX_TIP ];

@fernandojsg Where would I define `THREE.XRHand.INDEX_PHALANX_TIP`?

**Comment by cabanier** [comment]
@fernandojsg I decided to get rid of the indexes and move to using joint names.

**Comment by mrdoob** [comment]
Looking good!

> This patch should only be merged once the API goes live.

When do you think this would be?

**Comment by cabanier** [comment]
> Looking good!

Thanks!
> > This patch should only be merged once the API goes live.
> 
> When do you think this would be?

I believe sometime in January.

**Comment by mrdoob** [comment]
Okay! Let us know when it's out and I'll merge this 👍

**Comment by cabanier** [comment]
@mrdoob The new Oculus browser with the updated API is now rolling out to users.
Should we merge this PR?

**Comment by mrdoob** [comment]
Thanks!

File: src/renderers/webxr/WebXRController.js [hunk 1]
```diff
@@ -139,55 +123,64 @@ Object.assign( WebXRController.prototype, {
 
 				handPose = true;
 
-				for ( let i = 0; i <= window.XRHand.LITTLE_PHALANX_TIP; i ++ ) {
+				for ( const inputjoint of inputSource.hand.values() ) {
```
Comment (mrdoob) [inline src/renderers/webxr/WebXRController.js:126]: Is it possible to turn this into a simple `for` loop instead of a `for of` loop?

Comment (cabanier) [reply src/renderers/webxr/WebXRController.js:126]: What would be the advantage of doing it that way?

Comment (mrdoob) [reply src/renderers/webxr/WebXRController.js:126]: Avoiding having to add a polyfill for it:
https://github.com/mrdoob/three.js/commit/657b36d6132a9f522b11f6277e438d7171528a03#diff-af408426d6a7b072569230b0ee45fa87b6e69f3e97ebdc527d6995924aa1a1ceR1061-R1102

**Pull request merged**